### PR TITLE
Warnings report post process actions

### PR DESCRIPTION
**Pull request by nessid**
## Changelog Description
This pull request introduces a warnings report and post-process actions in the `ayon-core` project.
The primary change is the addition of the `PublishValidationException` class, designed to handle validation errors and warnings during the publishing process (it serves as a parent to the pre-existing `PublishValidationError` class and the new `PublishValidationWarning` class). 
Actions are now listed after the plugin process to incorporate dynamically added actions based on the process results.
Additionally, there are updates to color codes and UI elements to enhance visual clarity.

## Additional info
The `PublishValidationException` class is now the base for specific validation error and warning classes, offering detailed error messages and descriptions using Markdown syntax. This change enhances error handling and user communication in the publishing workflow. Actions listed after the plugin process ensure that any actions added dynamically based on the process results are captured and executed appropriately.

### CODE REVIEW

File: client/ayon_core/style/__init__.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ def get_objected_colors(*keys):
 
 
 def _load_stylesheet():
-    """Load strylesheet and trigger all related callbacks.
+    """Load stylesheet and trigger all related callbacks.
```
Comment (EnisNewborn) [inline client/ayon_core/style/__init__.py]: Not sure if it forth fork just for a description typo

File: client/ayon_core/tools/publisher/control.py [hunk 2]
```diff
@@ -423,7 +437,7 @@ class PublishPluginActionItem:
         action_id (str): Action id.
         plugin_id (str): Plugin id.
         active (bool): Action is active.
-        on_filter (str): Actions have 'on' attribte which define when can be
+        on_filter (str): Actions have 'on' attribute which define when can be
```
Comment (EnisNewborn) [inline client/ayon_core/tools/publisher/control.py]: Same. Not sure it worth fix source code

File: client/ayon_core/tools/publisher/control.py [hunk 3]
```diff
@@ -518,11 +534,12 @@ def to_data(self):
             "title": self.title,
             "description": self.description,
             "detail": self.detail,
+            "exception_type": self.exception_type,
         }
 
     @classmethod
     def from_result(cls, plugin_id, error, instance):
-        """Create new object based on resukt from controller.
+        """Create new object based on result from controller.
```
Comment (EnisNewborn) [inline client/ayon_core/tools/publisher/control.py]: Source code typo ;)

File: client/ayon_core/tools/publisher/widgets/images/thumb-up.png
Comment (EnisNewborn) [inline client/ayon_core/tools/publisher/widgets/images/thumb-up.png:1]: Please remove if not used 😁

File: client/ayon_core/tools/publisher/widgets/report_page.py [hunk 4]
```diff
@@ -1803,7 +1988,7 @@ def _on_error_selection(self):
 
 
 class ReportPageWidget(QtWidgets.QFrame):
-    """Widgets showing report for artis.
+    """Widgets showing report for artists.
```
Comment (EnisNewborn) [inline client/ayon_core/tools/publisher/widgets/report_page.py]: source typo ;)

**Review by EnisNewborn — COMMENTED** [review]
First short overview. 
I didn't realized the amount of work here. Great job !

File: client/ayon_core/style/__init__.py (hunk 1 above)
Comment (nessid) [reply client/ayon_core/style/__init__.py]: typo restored

File: client/ayon_core/tools/publisher/control.py (hunk 2 above)
Comment (nessid) [reply client/ayon_core/tools/publisher/control.py]: typo restored

File: client/ayon_core/tools/publisher/control.py (hunk 3 above)
Comment (nessid) [reply client/ayon_core/tools/publisher/control.py]: typo restored

File: client/ayon_core/tools/publisher/widgets/report_page.py (hunk 4 above)
Comment (nessid) [reply client/ayon_core/tools/publisher/widgets/report_page.py]: typo restored

File: client/ayon_core/tools/publisher/widgets/images/thumb-up.png
Comment (nessid) [reply client/ayon_core/tools/publisher/widgets/images/thumb-up.png:1]: image removed

**Comment by nessid** [comment]
This pull request is being closed because a new branch, `plugins_view_actions`, has been created based on it. The new branch includes the features from the `warnings-report_post-process-actions` branch as well as new functionality.

The `plugins_view_actions` branch introduces actions available in the plugins list in the "Details" tab of the publisher. This feature allows actions not only for plugins with errors or warnings but for all plugins with actions.

The `warnings-report_post-process-actions` branch featured the handling of non-blocking publish errors (warnings) in the report, with the ability to ignore them and proceed with publishing. It also included adding actions to `PublishPluginsProxy` after the plugins have been processed, to account for actions created dynamically during the process based on the results.

A new pull request will be created for the `plugins_view_actions` branch, with all changes from this pull request, along with the new features.

**Pull request closed without merging**